### ISSUE TITLE
Error deploying Wazuh Indexer cluster via Puppet 

### ISSUE THREAD

**Issue by Rolly-M** [wazuh/wazuh#19405]
|Wazuh version|Component|Install type|Install method|Platform|
|---|---|---|---|---|
| 4.6.0 Beta 1 | Wazuh component | Indexer | Puppet | Ubuntu 18.04|

I discovered in the test carried out in this [issue](https://github.com/wazuh/wazuh/issues/19150) that, deploying indexer cluster via puppet isn't successful. Each indexer node is installed and configured separately but are not part of the same cluster.

Testing the Wazuh indexer cluster with the security admin script provides the output below showing only one node:
```console
root@indexer2:/home/rolly# /usr/share/wazuh-indexer/bin/indexer-security-init.sh
**************************************************************************
** This tool will be deprecated in the next major release of OpenSearch **
** https://github.com/opensearch-project/security/issues/1755           **
**************************************************************************
Security Admin v7
Will connect to 127.0.0.1:9200 ... done
Connected as "CN=admin,OU=Wazuh,O=Wazuh,L=California,C=US"
OpenSearch Version: 2.8.0
Contacting opensearch cluster 'opensearch' and wait for YELLOW clusterstate ...
Clustername: wazuh-cluster
Clusterstate: GREEN
Number of nodes: 1
Number of data nodes: 1
.opendistro_security index does not exists, attempt to create it ... done (0-all replicas)
Populate config from /etc/wazuh-indexer/opensearch-security/
Will update '/config' with /etc/wazuh-indexer/opensearch-security/config.yml
   SUCC: Configuration for 'config' created or updated
Will update '/roles' with /etc/wazuh-indexer/opensearch-security/roles.yml
   SUCC: Configuration for 'roles' created or updated
Will update '/rolesmapping' with /etc/wazuh-indexer/opensearch-security/roles_mapping.yml
   SUCC: Configuration for 'rolesmapping' created or updated
Will update '/internalusers' with /etc/wazuh-indexer/opensearch-security/internal_users.yml
   SUCC: Configuration for 'internalusers' created or updated
Will update '/actiongroups' with /etc/wazuh-indexer/opensearch-security/action_groups.yml
   SUCC: Configuration for 'actiongroups' created or updated
Will update '/tenants' with /etc/wazuh-indexer/opensearch-security/tenants.yml
   SUCC: Configuration for 'tenants' created or updated
Will update '/nodesdn' with /etc/wazuh-indexer/opensearch-security/nodes_dn.yml
   SUCC: Configuration for 'nodesdn' created or updated
Will update '/whitelist' with /etc/wazuh-indexer/opensearch-security/whitelist.yml
   SUCC: Configuration for 'whitelist' created or updated
Will update '/audit' with /etc/wazuh-indexer/opensearch-security/audit.yml
   SUCC: Configuration for 'audit' created or updated
Will update '/allowlist' with /etc/wazuh-indexer/opensearch-security/allowlist.yml
   SUCC: Configuration for 'allowlist' created or updated
SUCC: Expected 10 config types for node {"updated_config_types":["allowlist","tenants","rolesmapping","nodesdn","audit","roles","whitelist","internalusers","actiongroups","config"],"updated_config_size":10,"message":null} is 10 (["allowlist","tenants","rolesmapping","nodesdn","audit","roles","whitelist","internalusers","actiongroups","config"]) due to: null
Done with success
```

Also although dashboard deployment was not part of the test description, I however tried to install but got the following error when trying to access via URL.

![image](https://github.com/wazuh/wazuh/assets/99815109/ef29cbf5-0120-4fd0-a874-55ea0f4077d1)

The dashboard error log shows the following:
```console
root@dashboard:/home/rolly# tail /tmp/wazuh-dashboard.log
Oct 04 11:33:15 dashboard opensearch-dashboards[6486]: {"type":"log","@timestamp":"2023-10-04T11:33:15Z","tags":["error","opensearch","data"],"pid":6486,"message":"[ConnectionError]: connect ECONNREFUSED 127.0.0.1:9200"}
Oct 04 11:33:17 dashboard opensearch-dashboards[6486]: {"type":"log","@timestamp":"2023-10-04T11:33:17Z","tags":["error","opensearch","data"],"pid":6486,"message":"[ConnectionError]: connect ECONNREFUSED 127.0.0.1:9200"}
Oct 04 11:33:20 dashboard opensearch-dashboards[6486]: {"type":"log","@timestamp":"2023-10-04T11:33:20Z","tags":["error","opensearch","data"],"pid":6486,"message":"[ConnectionError]: connect ECONNREFUSED 127.0.0.1:9200"}
Oct 04 11:33:22 dashboard opensearch-dashboards[6486]: {"type":"log","@timestamp":"2023-10-04T11:33:22Z","tags":["error","opensearch","data"],"pid":6486,"message":"[ConnectionError]: connect ECONNREFUSED 127.0.0.1:9200"}
Oct 04 11:33:25 dashboard opensearch-dashboards[6486]: {"type":"log","@timestamp":"2023-10-04T11:33:25Z","tags":["error","opensearch","data"],"pid":6486,"message":"[ConnectionError]: connect ECONNREFUSED 127.0.0.1:9200"}
Oct 04 11:33:27 dashboard opensearch-dashboards[6486]: {"type":"log","@timestamp":"2023-10-04T11:33:27Z","tags":["error","opensearch","data"],"pid":6486,"message":"[ConnectionError]: connect ECONNREFUSED 127.0.0.1:9200"}
Oct 04 11:33:30 dashboard opensearch-dashboards[6486]: {"type":"log","@timestamp":"2023-10-04T11:33:30Z","tags":["error","opensearch","data"],"pid":6486,"message":"[ConnectionError]: connect ECONNREFUSED 127.0.0.1:9200"}
Oct 04 11:33:32 dashboard opensearch-dashboards[6486]: {"type":"log","@timestamp":"2023-10-04T11:33:32Z","tags":["error","opensearch","data"],"pid":6486,"message":"[ConnectionError]: connect ECONNREFUSED 127.0.0.1:9200"}
Oct 04 11:33:35 dashboard opensearch-dashboards[6486]: {"type":"log","@timestamp":"2023-10-04T11:33:35Z","tags":["error","opensearch","data"],"pid":6486,"message":"[ConnectionError]: connect ECONNREFUSED 127.0.0.1:9200"}
Oct 04 11:33:37 dashboard opensearch-dashboards[6486]: {"type":"log","@timestamp":"2023-10-04T11:33:37Z","tags":["error","opensearch","data"],"pid":6486,"message":"[ConnectionError]: connect ECONNREFUSED 127.0.0.1:9200"}
```

The dashboard searches for the indexer on `localhost` instead of its IP address(`192.168.21.170` or `192.168.21.171` in the case of this test)

**Comment by teddytpc1** [comment]
Currently, the wazuh-puppet repository does not allow a distributed deployment.
It will be included in `4.8.0` with this issue:
- https://github.com/wazuh/wazuh-puppet/issues/632.

**Comment by teddytpc1** [comment]
We will continue the distributed deployment development here:
- https://github.com/wazuh/wazuh-puppet/issues/632
